Gems/Atom/RHI/Vulkan/Code/Source/RHI/RenderPass.cpp

```diff
@@ -16,6 +16,356 @@ namespace AZ
 {
     namespace Vulkan
     {
+        namespace Internal
+        {
+            enum class AttachmentType : uint32_t
+            {
+                Color, // Color render target attachment
+                DepthStencil, // Depth stencil attachment
+                Resolve, // Resolve attachment
+                InputAttachment, // An input attachment that is the output of a previous subpass.
+                Preserve, // An attachment that is not being used by the subpass but that it will be preserved.
+                ShadingRate, // An attachment that is used for specifiying shading rate.
+                Count
+            };
+            static const uint32_t AttachmentTypeCount = static_cast<uint32_t>(AttachmentType::Count);
+
+            using RenderPassResult = AZStd::tuple<VkResult, VkRenderPass>;
+
+            template<
+                typename VkRenderPassCreateInfoTraits,
+                typename VkAttachmentDescriptionTraits,
+                typename VkAttachmentReferenceTraits,
+                typename VkSubpassDescriptionTraits,
+                typename VkSubpassDependencyTraits>
+            struct VkRenderpassBuilder
```

Comment:
This struct needs a comment explaining why it is setup this way and what functionality it applies. I am assuming that the only new changes here are related to ShadingRate and everything else was copied over from before?